modules/builtin/src/hooks/after_incoming_middleware/00_dialog_engine.js

```diff
@@ -1,4 +1,4 @@
-const messageTypesToDiscard = ['session_reset', 'typing', 'visit', 'session_reference']
+const messageTypesToDiscard = ['session_reset', 'typing', 'visit', 'session_reference', 'proactive-trigger']
```

Comment:
Can you make sure this doesn't break the proactive triggers on the channel-web?

Reply:
If you check the doc it gives this example : 
```ts
// Catch the event
if (event.type === 'proactive-trigger') {
  const eventDestination = {
    channel: event.channel,
    target: event.target,
    botId: event.botId,
    threadId: event.threadId
  }

  // Skip event processing
  event.setFlag(bp.IO.WellKnownFlags.SKIP_DIALOG_ENGINE, true)

  // Make the bot respond with custom content instead
  bp.cms.renderElement('builtin_text', { text: "I'm so proactive!", typing: true }, eventDestination).then(payloads => {
    bp.events.replyToEvent(event, payloads)
  })
}
```
So normally proactive-trigger events should ski the dialog engine. It's possible someone has proactive-trigger events that don't skip the dialog engine. What we could do is require to manually add the SKIP_DIALOG_ENGINE flag like the tutorial says

Reply:
Yeah I would go that way to make sure we are not breaking for anyone